src/osp/Active/SysWire.cpp

```diff
@@ -27,32 +27,53 @@
 #include "SysWire.h"
 #include "ActiveScene.h"
 
-using namespace osp;
-using namespace osp::active;
+using osp::active::SysWire;
 
-WireData* WireInput::connected_value()
+void SysWire::add_functions(ActiveScene& rScene)
 {
-    WireOutput* woConnected = list();
-    if (woConnected)
-    {
-        return &(woConnected->value());
-    }
-    else
-    {
-        return nullptr;
-    }
+    rScene.debug_update_add(rScene.get_update_order(), "wire", "vehicle_modification", "physics",
+                            &SysWire::update_wire);
 }
 
-void WireInput::doErase()
+void SysWire::setup_default(
+        ActiveScene& rScene,
+        std::vector<ACompWire::update_func_t> updCalculate,
+        std::vector<ACompWire::update_func_t> updPropagate)
 {
-     list()->cut(this);
-};
-
+    rScene.reg_emplace<ACompWire>(rScene.hier_get_root(),
+                                  ACompWire{std::move(updCalculate),
+                                            std::move(updPropagate)});
+}
 
-void SysWire::connect(WireOutput &wireFrom, WireInput &wireTo)
+void SysWire::update_wire(ActiveScene &rScene)
 {
-  SPDLOG_LOGGER_INFO(spdlog::get("application"), "Connected {} to {}", wireFrom.get_name(),
-                      wireTo.get_name());
-    wireFrom.insert(&wireTo);
-    // TODO: check for dependent outputs and add to list and sort
+    auto view = rScene.get_registry().view<ACompWireNeedUpdate>();
+
+    ActiveReg_t::poly_storage t = rScene.get_registry().storage(entt::type_id<ACompWireNeedUpdate>());
+
+
+    auto const& wire = rScene.reg_get<ACompWire>(rScene.hier_get_root());
+
+    int updateLimit = 16;
```

Comment:
Recommend against signed. And recommend explicitly specifying bit width.

Reply:
This doesn't seem that important. It's a simple count that won't go any higher than the value assigned to it now, and it's also in a temporary place where it should be made configurable from elsewhere eventually.